---
title: "RNAseq viewer report"
author: "`r params$author`"
output: 
    flexdashboard::flex_dashboard:
        vertical_layout: scroll
        orientation: rows
params:
  nr: NA
  bpnr: NA
  ccnr: NA
  mfnr: NA
  nrdown: NA
  bpnrdown: NA
  mfnrdown: NA
  ccnrdown: NA
  nrall: NA
  bpnrall: NA
  mfnrall: NA
  ccnrall: NA
  variablepca: NA
  tempdir: NA
  gseanr: NA
  author: NA
---

```{r results="asis", echo=FALSE}
cat("
<style>
.navbar-inverse .navbar-nav>li>a {
    color: #ffffff;
    background-color: #334969;
}

.navbar-inverse .navbar-nav>.active>a, .navbar-inverse .navbar-nav>.active>a:hover, .navbar-inverse .navbar-nav>.active>a:focus {
    color: #ffffff;
    background-color: #3b5e86;
}

.navbar-inverse .navbar-collapse, .navbar-inverse .navbar-form {
    border-color: #396596;
    background-color: #334969;
}

.container-fluid {
    background-color: #334969;
}

.navbar-inverse .navbar-nav > .open > a, .navbar-inverse .navbar-nav > .open > a:hover, .navbar-inverse .navbar-nav > .open > a:focus {
    background-color: #3b5e86;
    color: #ffffff;

}

.dropdown-menu > .active > a, .dropdown-menu > .active > a:hover, .dropdown-menu > .active > a:focus {
    color: #ffffff;
    text-decoration: none;
    outline: 0;
    background-color: #3b5e86;
}

.navbar-inverse .navbar-nav > li > a:hover, .navbar-inverse .navbar-nav > li > a:focus {

    color: #ffffff;
    background-color: #3b5e86;
}

.navbar-inverse .navbar-nav > .open > a, .navbar-inverse .navbar-nav > .open > a:hover, .navbar-inverse .navbar-nav > .open > a:focus {

    background-color: #3b5e86;
    color: #ffffff;
}
.dropdown-menu > li > a:hover, .dropdown-menu > li > a:focus {
    text-decoration: none;
    color: #ffffff;
    background-color: #3b5e86;
}
</style>
")
```

```{r setup, echo=FALSE, include=FALSE, comment=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, fig.width = 12, eval=TRUE)

library('dygraphs')
library('flexdashboard')
library('shinydashboard')
# library('limma')
# library('tidyverse')
# library('DT')
# library('purrr')
# library('plotly')
# library('DESeq2')
# library("ggraph")
# library("igraph")
# source("utils.R")
```

```{r}
setwd(params$tempdir)
genesUp <- readRDS( "tmpResources/genesUp.Rds")
genesDown <- readRDS("tmpResources/genesDown.Rds")
genesall <- readRDS("tmpResources/kggDTall.Rds")
kggUp <- readRDS( "tmpResources/kggUp.Rds")
kggDTup <- readRDS( "tmpResources/kggDTup.Rds")
goUp <- readRDS( "tmpResources/goUp.Rds")
goDTup <- readRDS( "tmpResources/goDTup.Rds")
kggDown <- readRDS( "tmpResources/kggDown.Rds")
kggDTdown <- readRDS( "tmpResources/kggDTdown.Rds")
goDown <- readRDS( "tmpResources/goDown.Rds")
goDTdown <- readRDS( "tmpResources/goDTdown.Rds")
dds <- readRDS("tmpResources/dds.Rds")
gsea <- readRDS("tmpResources/gsea.Rds")
kggAll <- readRDS( "tmpResources/kggAll.Rds")
kggDTall <- readRDS( "tmpResources/kggDTall.Rds")
```



Data preview
=====================================

Row
--------------------------------------

### Biological context

```{r results="asis", eval=TRUE}

texto <- "
* Samples background: CE1502 - Astrocytes WT with two different AAV (GFP vs SOCS3) extracted from the Kelly AD project.
* Raw data (Fasta files): /local/aa_SideProject_ComparisonWT/FASTAfiles
* Bam files
    * Aligned Genosplice (STAR 2.4): /home/Pactivastro/Genosplice-BAMfiles (All) or /local/aa_SideProject_ComparisonWT/BAMstar (subset)

–outSAMstrandField intronMotif –outFilterMismatchNmax 2 –outFilterMultimapNmax 10 –outSAMunmapped Within –outSAMtype BAM SortedByCoordinate –outStd BAM_SortedByCoordinate –genomeLoad NoSharedMemory –chimSegmentMin 15

* Aligned myself (STAR with customed genome adding GFPV5): /local/aa_SideProject_ComparisonWT/CUSTOM
* Aligned myself (HISAT2): /local/aa_SideProject_ComparisonWT/BAMhisat2
* Folder routes for the whole project
    * Local linux: /local/aa_SideProject_ComparisonWT (Full data files)
    * Pactivastro: /home/Pactivastro/Resultats-manip/Miriam/aa_SideProject_ComparisonWT (Only reports)
"
cat(texto)
```

Row
--------------------------------------

### Metata experiment {data-width=500}

```{r}

metadata <- as.data.frame(colData(dds)) %>% select(-c(sizeFactor,replaceable))
      datatable( metadata, 
                 rownames=FALSE,
                 filter = list(position="top", clear=FALSE),
                 options = list(
                   columnDefs = list(list(orderable = FALSE,
                                          className = "details-control",
                                          targets = 1),
                                     list(className = "dt-right", targets = 1:(ncol(metadata)-1))
                   ),
                   dom = "Bfrtipl",
                   buttons = c("copy", "csv", "excel", "pdf", "print"),
                   list(pageLength = 10, white_space = "normal")
                 )
      )
```

### Especific comment {data-width=500}

```{r results="asis"}

```

Row
--------------------------------------

### DEseq results 

```{r}
res <- results(dds)
res <- as.data.frame(res)
res <- res[res$padj<=0.05, ]
conversion <- geneIdConverter(rownames(res))
res <- round(res,4)
res <- cbind(`Gene name`=conversion$consensus, res)
#add_column(res, Symbol=conversion$consensus, .before = "baseMean")
datatable( res, 
           filter = list(position="top", clear=FALSE),
           options = list(
               columnDefs = list(list(orderable = FALSE,
                                      className = "details-control",
                                      targets = 1),
                                 list(className = "dt-right", targets = 1:ncol(res))
               ),
               rowCallback = JS(
                   "function(row, data) {",
                   "for (i = 6; i < 8; i++) {",
                   "if (data[i]>1000 | data[i]<1){",
                   "$('td:eq('+i+')', row).html(data[i].toExponential(3));",
                   "}",
                   "}",
                   "}"),
               dom = "Bfrtipl",
               buttons = c("copy", "csv", "excel", "pdf", "print"),
               list(pageLength = 10, white_space = "normal")
           )
)
```


### PCA gene expresión 

```{r out.width="800px", out.height="800px"}
plotPCA(rlog(dds), intgroup = params$variablepca )+
            #theme(plot.margin=unit(c(0.5,0.5,0.5,0.5),"cm"))+
            theme(text = element_text(size=14))
```


All together {data-navmenu="Kegg"}
=====================================

Row
------------------------------------

### Kyoto Encyclopedia of Genes and Genomes (KEGG)

KEGG is a database resource for understanding high-level functions and utilities of the biological system, such as the cell, the organism and the ecosystem, from molecular-level information, especially large-scale molecular datasets generated by genome sequencing and other high-throughput experimental technologies.

Especially, KEGG PATHWAY is a collection of manually drawn pathway maps representing our knowledge on the molecular interaction, reaction and relation networks for:

* Metabolism:
    * Global/overview Carbohydrate Energy Lipid Nucleotide Amino acid Other amino Glycan
    * Cofactor/vitamin Terpenoid/PK Other secondary metabolite Xenobiotics Chemical structure
* Genetic Information Processing.
* Environmental Information Processing.
* Cellular Processes.
* Organismal Systems.
* Human Diseases.
* Drug Development.

Row 
------------------------------------

### tabla {data-width=500}

```{r echo=FALSE}
predata <- kegg2DT(kggAll, genesall, nrows = NULL)
datatable2( predata,
            vars = c("genes"),
            escape = FALSE,
            opts = list(pageLength = 10, white_space = "normal"))
```


### Visualization of all pathways {data-width=500}

This table here is similar to the previous one, with the pathways instead of GO terms. Also, take a look in the extra option, when you click in the code in blue (mmu00000) for every row, that allows you to access the pathway map in KEGG and highlight in red those genes within the route that have appeared in our list.

Row
------------------------------------

### Barplot

```{r  echo=FALSE}
plotKegg(enrichdf = kggUp[params$nrall,], nrows = length(params$nrall) ) 
```

### ChorPlot

```{r echo=FALSE, warning=FALSE}
chordPlot(kggUp[params$nrall, ], nRows = length(params$nrall), orderby = "P.DE")
```

Row
------------------------------------

### Dotplot

```{r}
dotPlotkegg(kggUp[params$nrall,], n = length(params$nrall))+
        theme(text = element_text(size=10))
```


### Kegg heatmap

```{r}
heatmapKegg(kggDTall, params$nrall)
```

Row
------------------------------------

### Kegg cnetplot

```{r}
customCnetKegg(kggAll, params$nrall)
```

### 

```{r}

```


Upregulated {data-navmenu="Kegg"}
=====================================  

Row
------------------------------------

### text 

```{r}

```


Row 
------------------------------------

### tabla {data-width=500}

```{r echo=FALSE}

```


### text1 {data-width=500}

```{r}

```

Row 
------------------------------------

### Lorem ipsum {data-width=500}

Lorem ipsum is simply dummy text of the printing and typesetting industry.
Lorem Ipsum has been the industry's standard dummy text ever since the 1500s,
when an unknown printer took a galley of type and scrambled it to make a type
specimen book. It has survived not only five centuries, but also the leap into
electronic typesetting, remaining essentially unchanged. It was popularised in
the 1960s with the release of Letraset sheets containing Lorem Ipsum passages,
and more recently with desktop publishing software like Aldus PageMaker
including versions of Lorem Ipsum.

* asñlfk
* sdfs
* sdfds

### text2 {data-width=500}

Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.


Row
------------------------------------

### Barplot

```{r  echo=FALSE}

```

### ChorPlot

```{r echo=FALSE, warning=FALSE}

```

Row
------------------------------------

### Dotplot

```{r}
dotPlotkegg(kggUp[params$nr,], n = length(params$nr))+
        theme(text = element_text(size=10))
```


### Kegg heatmap

```{r}

```

Row
------------------------------------

### Kegg cnetplot

```{r}

```

### 

```{r}

```


Downregulated {data-navmenu="Kegg"}
=====================================

Row
------------------------------------

### text 

```{r}

```


Row 
------------------------------------

### tabla {data-width=500}

```{r echo=FALSE}

```


### text1 {data-width=500}

```{r}

```

Row 
------------------------------------

### Lorem ipsum {data-width=500}

Lorem ipsum is simply dummy text of the printing and typesetting industry.
Lorem Ipsum has been the industry's standard dummy text ever since the 1500s,
when an unknown printer took a galley of type and scrambled it to make a type
specimen book. It has survived not only five centuries, but also the leap into
electronic typesetting, remaining essentially unchanged. It was popularised in
the 1960s with the release of Letraset sheets containing Lorem Ipsum passages,
and more recently with desktop publishing software like Aldus PageMaker
including versions of Lorem Ipsum.

* asñlfk
* sdfs
* sdfds

### text2 {data-width=500}

Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.


Row
------------------------------------

### Barplot

```{r  echo=FALSE}

```

### ChorPlot

```{r echo=FALSE, warning=FALSE}

```

Row
------------------------------------

### Dotplot

```{r}

```


### Kegg heatmap

```{r}

```

Row
------------------------------------

### Kegg cnetplot

```{r}

```

### 

```{r}

```


All together {data-navmenu="GO BP"}
=====================================

Row
------------------------------------

### text 

```{r}

```


Row 
------------------------------------

### tabla {data-width=500}

```{r echo=FALSE}

```


### text1 {data-width=500}

```{r}

```

Row 
------------------------------------

### Lorem ipsum {data-width=500}

Lorem ipsum is simply dummy text of the printing and typesetting industry.
Lorem Ipsum has been the industry's standard dummy text ever since the 1500s,
when an unknown printer took a galley of type and scrambled it to make a type
specimen book. It has survived not only five centuries, but also the leap into
electronic typesetting, remaining essentially unchanged. It was popularised in
the 1960s with the release of Letraset sheets containing Lorem Ipsum passages,
and more recently with desktop publishing software like Aldus PageMaker
including versions of Lorem Ipsum.

* asñlfk
* sdfs
* sdfds

### text2 {data-width=500}

Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.


Row
------------------------------------

### Barplot {data-width=500}

```{r  echo=FALSE}

```

### ChorPlot {data-width=500}

```{r echo=FALSE, warning=FALSE}

```

Row
------------------------------------

### Dotplot {data-width=500}

```{r}

```


### Kegg heatmap {data-width=500}

```{r}

```

Row
------------------------------------

### Kegg cnetplot {data-width=500}

```{r}
customCnetKegg(kggUp, params$nr)
```

### {data-width=500}

```{r}

```

Upregulated {data-navmenu="GO BP"}
=====================================

Downregulated {data-navmenu="GO BP"}
=====================================

All together {data-navmenu="GO MF"}
=====================================

Upregulated {data-navmenu="GO MF"}
=====================================

Downregulated {data-navmenu="GO MF"}
=====================================

All together {data-navmenu="GO CC"}
=====================================

Upregulated {data-navmenu="GO CC"}
=====================================

Downregulated {data-navmenu="GO CC"}
=====================================

GSEA
=====================================

Row
-------------------------------------

### Generic and specific text

```{r}

```

Row
------------------------------------

### Table GSEA {data-width=500}

```{r}

```

### Plot GSEA {data-width=500}

```{r}

```

