---
title: "RNAseq viewer report"
author: "`r params$author`"
output: 
    flexdashboard::flex_dashboard:
        vertical_layout: scroll
        orientation: rows
params:
  nrup: NA
  bpnrup: NA
  ccnrup: NA
  mfnrup: NA
  nrdown: NA
  bpnrdown: NA
  mfnrdown: NA
  ccnrdown: NA
  nrall: NA
  bpnrall: NA
  mfnrall: NA
  ccnrall: NA
  variablepca: NA
  tempdir: NA
  gseanr: NA
  author: NA
  biologicalText: NA
  explainPreview: NA
  keggAllText: NA
  deseq: NA
  kggUp: NA
  kggAll: NA
  kggDown: NA
  kggDTall: NA
  kggDTdown: NA
  kggDTup: NA
  goAll: NA
  goUp: NA
  goDown: NA
  goDTall: NA
  goDTdown: NA
  goDTup: NA
  gsea: NA
  
---

```{r results="asis", echo=FALSE}
cat("
<style>
.navbar-inverse .navbar-nav>li>a {
    color: #ffffff;
    background-color: #334969;
}

.navbar-inverse .navbar-nav>.active>a, .navbar-inverse .navbar-nav>.active>a:hover, .navbar-inverse .navbar-nav>.active>a:focus {
    color: #ffffff;
    background-color: #3b5e86;
}

.navbar-inverse .navbar-collapse, .navbar-inverse .navbar-form {
    border-color: #396596;
    background-color: #334969;
}

.container-fluid {
    background-color: #334969;
}

.navbar-inverse .navbar-nav > .open > a, .navbar-inverse .navbar-nav > .open > a:hover, .navbar-inverse .navbar-nav > .open > a:focus {
    background-color: #3b5e86;
    color: #ffffff;

}

.dropdown-menu > .active > a, .dropdown-menu > .active > a:hover, .dropdown-menu > .active > a:focus {
    color: #ffffff;
    text-decoration: none;
    outline: 0;
    background-color: #3b5e86;
}

.navbar-inverse .navbar-nav > li > a:hover, .navbar-inverse .navbar-nav > li > a:focus {

    color: #ffffff;
    background-color: #3b5e86;
}

.navbar-inverse .navbar-nav > .open > a, .navbar-inverse .navbar-nav > .open > a:hover, .navbar-inverse .navbar-nav > .open > a:focus {

    background-color: #3b5e86;
    color: #ffffff;
}
.dropdown-menu > li > a:hover, .dropdown-menu > li > a:focus {
    text-decoration: none;
    color: #ffffff;
    background-color: #3b5e86;
}

#section-1.section.level3.chart-wrapper.chart-wrapper-flex{
    background: #f2f2f2;
    border: #f2f2f2
}
#section.section.level3.chart-wrapper.chart-wrapper-flex{
    background: #f2f2f2;
    border: #f2f2f2
}
#enrich-app-2020.section.level4{
    text-align: center;
}
</style>
")
```

```{r setup, echo=FALSE, include=FALSE, comment=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, fig.width = 12, eval=TRUE)

library('dygraphs')
library('flexdashboard')
library('shinydashboard')
library('limma')
library('tidyverse')
library('DT')
library('purrr')
library('plotly')
library('DESeq2')
library("ggraph")
library("igraph")
source("utils.R")
```

```{r}
setwd(params$tempdir)
#genesUp <- readRDS( "tmpResources/genesUp.Rds")
#genesDown <- readRDS("tmpResources/genesDown.Rds")
#genesall <- readRDS("tmpResources/kggDTall.Rds")
#kggUp <- readRDS( "tmpResources/kggUp.Rds")
#kggDTup <- readRDS( "tmpResources/kggDTup.Rds")
#goUp <- readRDS( "tmpResources/goUp.Rds")
#goDTup <- readRDS( "tmpResources/goDTup.Rds")
#kggDown <- readRDS( "tmpResources/kggDown.Rds")
#kggDTdown <- readRDS( "tmpResources/kggDTdown.Rds")
#goDown <- readRDS( "tmpResources/goDown.Rds")
#goDTdown <- readRDS( "tmpResources/goDTdown.Rds")
#deseq <- readRDS("tmpResources/dds.Rds")
#gsea <- readRDS("tmpResources/gsea.Rds")
#kggAll <- readRDS( "tmpResources/kggAll.Rds")
#kggDTall <- readRDS( "tmpResources/kggDTall.Rds")
```



Data preview
=====================================

Row {data-height=250}
--------------------------------------

### Biological context 

```{r results="asis"}
cat(params$biologicalText)
```

Row
--------------------------------------

### Metadata experiment {data-width=500}

```{r}
    metadata <- as.data.frame(colData(params$deseq)) %>% select(-c(sizeFactor,replaceable))
    datatable( metadata, extensions = "Buttons",
               rownames=FALSE,
               filter = list(position="top", clear=FALSE),
               options = list(
                 columnDefs = list(list(orderable = TRUE,
                                        className = "details-control",
                                        targets = 1),
                                   list(className = "dt-right", targets = 1:(ncol(metadata)-1))
                 ),
                 dom = "Bfrtipl",
                 buttons = c("copy", "csv", "excel", "pdf", "print"),
                 list(pageLength = 10, white_space = "normal")
               )
    )
```

### DEseq results {data-width=500}

```{r results="asis"}
res <- as.data.frame(lfcShrink(params$deseq, coef=2, type="apeglm", res = results(params$deseq)))
    conversion <- geneIdConverter(rownames(res))
    res$baseMean <- round(res$baseMean,4)
    res$lfcSE <- round(res$lfcSE,4)
    res$log2FoldChange <- round(res$log2FoldChange,4)
    res <- cbind(`Description`=conversion$description, res)
    res <- cbind(`GeneName/Symbol`=conversion$consensus, res)
    res <-  res %>% select(-c(pvalue))
    links = paste0("<a href='http://www.ensembl.org/Mus_musculus/Gene/Summary?db=core;g=",
                   rownames(res),"' target='_blank'>",rownames(res),"</a>")
    res <- cbind(`GeneName/Ensembl`= links, res)
    
    datatable( res, extensions = "Buttons", escape = FALSE,
               rownames = FALSE,
               filter = list(position="top", clear=FALSE),
               options = list(
                 columnDefs = list(list(orderable = FALSE,
                                        className = "details-control",
                                        targets = 1),
                                   list(className = "dt-right", targets = 1:(ncol(res)-1))
                 ),
                 rowCallback = JS(
                   "function(row, data) {",
                   "for (i = 6; i < 8; i++) {",
                   "if (data[i]>1000 | data[i]<1){",
                   "$('td:eq('+i+')', row).html(data[i].toExponential(3));",
                   "}",
                   "}",
                   "}"),
                 dom = "Bfrtipl",
                 buttons = c("copy", "csv", "excel", "pdf", "print"),
                 list(pageLength = 10, white_space = "normal")
               )
    )
```

Row {data-height=500}
--------------------------------------

### Samples PCA gene expression {data-width=100}

```{r fig.asp=0.5}
plotPCA(rlog(params$deseq), intgroup = params$variablepca ) +
  theme(plot.margin=unit(c(0.5,0.5,0.5,0.5),"cm"))+
  scale_size_manual(values = 4) +
  theme(text = element_text(size=16))

```


### Preview background {data-width=50}

```{r results='asis'}
cat(params$explainPreview)
```


All together {data-navmenu="Kegg"}
=====================================

Row {data-height=250}
------------------------------------

### Kyoto Encyclopedia of Genes and Genomes (KEGG) {data-width=500}

KEGG is a database resource for understanding high-level functions and utilities of the biological system, such as the cell, the organism and the ecosystem, from molecular-level information, especially large-scale molecular datasets generated by genome sequencing and other high-throughput experimental technologies.

Especially, KEGG PATHWAY is a collection of manually drawn pathway maps representing our knowledge on the molecular interaction, reaction and relation networks for:

* Metabolism:
    * Global/overview Carbohydrate Energy Lipid Nucleotide Amino acid Other amino Glycan
    * Cofactor/vitamin Terpenoid/PK Other secondary metabolite Xenobiotics Chemical structure
* Genetic Information Processing.
* Environmental Information Processing.
* Cellular Processes.
* Organismal Systems.
* Human Diseases.
* Drug Development.

### All genes {data-width=500}

```{r results='asis'}
cat(params$keggAllText)
```


Row 
------------------------------------

### Kegg Enrichment results --All genes-- {data-width=500}

```{r echo=FALSE}
predata <- params$kggDTall
datatable2(
      predata,
      vars = c("genes"),
      filter = list(position="top", clear=FALSE),
      escape = FALSE,
      opts = list(pageLength = 10, white_space = "normal"))
```

### Barplot --all genes-- {data-width=500}

```{r  echo=FALSE}
plotKegg(enrichdf = params$kggAll[params$nrall,], nrows = length(params$nrall) ) 
```

Row
------------------------------------

### ChorPlot --all genes-- {data-width=500}

```{r echo=FALSE, warning=FALSE}
chordPlot(params$kggAll[params$nrall, ], nRows = length(params$nrall), orderby = "P.DE")
```


### Dotplot --all genes-- {data-width=500}

```{r}
dotPlotkegg(params$kggAll[params$nrall,], n = length(params$nrall))+
        theme(text = element_text(size=10))
```

Row
------------------------------------

### Kegg heatmap --all genes-- {data-width=500}

```{r}
heatmapKegg(params$kggDTall, params$nrall)
```


### Kegg cnetplot --all genes-- {data-width=500}

```{r}
customCnetKegg(params$kggAll, params$nrall)
```


Upregulated genes {data-navmenu="Kegg"}
=====================================

Row {data-height=250}
------------------------------------

### Kyoto Encyclopedia of Genes and Genomes (KEGG) {data-width=500}

KEGG is a database resource for understanding high-level functions and utilities of the biological system, such as the cell, the organism and the ecosystem, from molecular-level information, especially large-scale molecular datasets generated by genome sequencing and other high-throughput experimental technologies.

Especially, KEGG PATHWAY is a collection of manually drawn pathway maps representing our knowledge on the molecular interaction, reaction and relation networks for:

* Metabolism:
    * Global/overview Carbohydrate Energy Lipid Nucleotide Amino acid Other amino Glycan
    * Cofactor/vitamin Terpenoid/PK Other secondary metabolite Xenobiotics Chemical structure
* Genetic Information Processing.
* Environmental Information Processing.
* Cellular Processes.
* Organismal Systems.
* Human Diseases.
* Drug Development.

### Upregulated genes {data-width=500}

```{r results='asis'}
cat("#### Texto variable")
```


Row 
------------------------------------

### Kegg Enrichment results --Upregulated genes-- {data-width=500}

```{r echo=FALSE}

predata <- params$kggDTup
    datatable2(
      predata,
      vars = c("genes"),
      filter = list(position="top", clear=FALSE),
      escape = FALSE,
      opts = list(pageLength = 10, white_space = "normal"))

```

### Barplot --Upregulated genes-- {data-width=500}

```{r  echo=FALSE}
plotKegg(enrichdf = params$kggUp[params$nrup,], nrows = length(params$nrup))
```

Row
------------------------------------

### ChorPlot --Upregulated genes-- {data-width=500}

```{r echo=FALSE, warning=FALSE}
chordPlot(params$kggUp[params$nrup, ], nRows = length(params$nrup), orderby = "P.DE")
```


### Dotplot --Upregulated genes-- {data-width=500}

```{r}
dotPlotkegg(params$kggUp[params$nrup,], n = length(params$nrup))
```

Row
------------------------------------

### Kegg heatmap --Upregulated genes-- {data-width=500}

```{r}
heatmapKegg(params$kggDTup, params$nrup)
```


### Kegg cnetplot --Upregulated genes-- {data-width=500}

```{r}
customCnetKegg(params$kggUp, params$nrup)
```


Downregulated genes {data-navmenu="Kegg"}
=====================================

Row {data-height=250}
------------------------------------

### Kyoto Encyclopedia of Genes and Genomes (KEGG) {data-width=500}

KEGG is a database resource for understanding high-level functions and utilities of the biological system, such as the cell, the organism and the ecosystem, from molecular-level information, especially large-scale molecular datasets generated by genome sequencing and other high-throughput experimental technologies.

Especially, KEGG PATHWAY is a collection of manually drawn pathway maps representing our knowledge on the molecular interaction, reaction and relation networks for:

* Metabolism:
    * Global/overview Carbohydrate Energy Lipid Nucleotide Amino acid Other amino Glycan
    * Cofactor/vitamin Terpenoid/PK Other secondary metabolite Xenobiotics Chemical structure
* Genetic Information Processing.
* Environmental Information Processing.
* Cellular Processes.
* Organismal Systems.
* Human Diseases.
* Drug Development.

### Downregulated genes {data-width=500}

```{r}

```


Row 
------------------------------------

### Kegg Enrichment results --Downregulated genes-- {data-width=500}

```{r echo=FALSE}
predata <- params$kggDTdown
    datatable2(
      predata,
      vars = c("genes"),
      filter = list(position="top", clear=FALSE),
      escape = FALSE,
      opts = list(pageLength = 10, white_space = "normal"))
```

### Barplot --Downregulated genes-- {data-width=500}

```{r  echo=FALSE}
plotKegg(enrichdf = params$kggDown[params$nrdown,], nrows = length(params$nrdown))
```

Row
------------------------------------

### ChorPlot --Downregulated genes-- {data-width=500}

```{r echo=FALSE, warning=FALSE}
chordPlot(params$kggDown[params$nrdown, ], nRows = length(params$nrdown), orderby = "P.DE")
```


### Dotplot --Downregulated genes-- {data-width=500}

```{r}
dotPlotkegg(params$kggDown[params$nrdown,], n = length(params$nrdown))
```

Row
------------------------------------

### Kegg heatmap --Downregulated genes-- {data-width=500}

```{r}
heatmapKegg(params$kggDTdown, params$nrdown)
```


### Kegg cnetplot --Downregulated genes-- {data-width=500}

```{r}
customCnetKegg(params$kggDown, params$nrdown)
```






Biological Process {data-navmenu="GO all together"}
=====================================

Row {data-height=250}
------------------------------------

### Gene Onthology (GO) : Biological process   {data-width=500}

Annotation related to the **biological process (BP)**: The larger processes, or ‘biological programs’ accomplished by multiple molecular activities. Note that a biological process is not equivalent to a pathway. At present, the GO does not try to represent the dynamics or dependencies that would be required to fully describe a pathway.


Row {data-height=500}
------------------------------------

### GO Enrichment results --All genes-- {data-width=500}

```{r}
goDTall <- params$goDTall
names(goDTall)[names(goDTall) == "level"] <- "Ont.level"
    goDTall$Ont.level = as.integer(goDTall$Ont.level) 
    datatable2(goDTall[goDTall$Ont=="BP",], vars = c("genes"),
               filter = list(position="top", clear=FALSE),
               escape = FALSE,
               opts = list(pageLength = 10, white_space = "normal")
    )
```

### Texto variable {data-width=500}

```{r results='asis'}


```

Row {data-height=500}
------------------------------------

### GO barplot --All genes-- {data-width=500}

```{r}
gosBP <- params$goAll[params$goAll$Ont == "BP",]
plotGO(enrichdf = gosBP[params$bpnrall, ], nrows = length(params$bpnrall), ont="BP")

```

### GO dotplot --All genes-- {data-width=500}

```{r}
gosBP <- params$goAll[params$goAll$Ont == "BP",]
dotPlotGO(gosBP[params$bpnrall,], n = length(params$bpnrall))
```


Molecular function {data-navmenu="GO all together"}
=====================================

Row {data-height=250}
------------------------------------

### Gene Onthology (GO) : Molecular function   {data-width=500}

Annotation related to the **molecular function (MF)**: Molecular-level activities performed by gene products. Describe activities that occur at the molecular level, such as “catalysis” or “transport”. GO molecular function terms represent activities rather than the entities (molecules or complexes) that perform the actions, and do not specify where, when, or in what context the action takes place. Molecular functions generally correspond to activities that can be performed by individual gene products (i.e. a protein or RNA), but some activities are performed by molecular complexes composed of multiple gene products.

Row {data-height=500}
------------------------------------

### GO Enrichment results --All genes-- {data-width=500}

```{r}
goDTall <- params$goDTall
names(goDTall)[names(goDTall) == "level"] <- "Ont.level"
    goDTall$Ont.level = as.integer(goDTall$Ont.level) 
    datatable2(goDTall[goDTall$Ont=="MF",], vars = c("genes"),
               filter = list(position="top", clear=FALSE),
               escape = FALSE,
               opts = list(pageLength = 10, white_space = "normal")
    )
```

### Texto variable {data-width=500}

```{r results='asis'}


```

Row {data-height=500}
------------------------------------

### GO barplot --All genes-- {data-width=500}

```{r}
gosMF <- params$goAll[params$goAll$Ont == "MF",]
plotGO(enrichdf = gosMF[params$mfnrall, ], nrows = length(params$mfnrall), ont="MF")

```

### GO dotplot --All genes-- {data-width=500}

```{r}
gosMF <- params$goAll[params$goAll$Ont == "MF",]
dotPlotGO(gosMF[params$mfnrall,], n = length(params$mfnrall))
```



Cellular component {data-navmenu="GO all together"}
=====================================

Row {data-height=250}
------------------------------------

### Gene Onthology (GO) : Cellular component   {data-width=500}

Annotation related to the **celular component (CC)**: The locations relative to cellular structures in which a gene product performs a function, either cellular compartments or stable macromolecular complexes of which they are parts.

Row {data-height=500}
------------------------------------

### GO Enrichment results --All genes-- {data-width=500}

```{r}
goDTall <- params$goDTall
names(goDTall)[names(goDTall) == "level"] <- "Ont.level"
    goDTall$Ont.level = as.integer(goDTall$Ont.level) 
    datatable2(goDTall[goDTall$Ont=="CC",], vars = c("genes"),
               filter = list(position="top", clear=FALSE),
               escape = FALSE,
               opts = list(pageLength = 10, white_space = "normal")
    )
```

### Texto variable {data-width=500}

```{r results='asis'}


```

Row {data-height=500}
------------------------------------

### GO barplot --All genes-- {data-width=500}

```{r}
gosCC <- params$goAll[params$goAll$Ont == "CC",]
plotGO(enrichdf = gosCC[params$ccnrall, ], nrows = length(params$ccnrall), ont="CC")

```

### GO dotplot --All genes-- {data-width=500}

```{r}
gosCC <- params$goAll[params$goAll$Ont == "CC",]
dotPlotGO(gosCC[params$ccnrall,], n = length(params$ccnrall))
```




Biological Process {data-navmenu="GO Upregulated"}
=====================================

Row {data-height=250}
------------------------------------

### Gene Onthology (GO) : Biological process   {data-width=500}

Annotation related to the **biological process (BP)**: The larger processes, or ‘biological programs’ accomplished by multiple molecular activities. Note that a biological process is not equivalent to a pathway. At present, the GO does not try to represent the dynamics or dependencies that would be required to fully describe a pathway.


Row {data-height=500}
------------------------------------

### GO Enrichment results --Upregulated genes-- {data-width=500}

```{r}
goDTup <- params$goDTup
names(goDTup)[names(goDTup) == "level"] <- "Ont.level"
    goDTup$Ont.level = as.integer(goDTup$Ont.level) 
    datatable2(goDTup[goDTup$Ont=="BP",], vars = c("genes"),
               filter = list(position="top", clear=FALSE),
               escape = FALSE,
               opts = list(pageLength = 10, white_space = "normal")
    )
```

### Texto variable {data-width=500}

```{r results='asis'}


```

Row {data-height=500}
------------------------------------

### GO barplot --Upregulated genes-- {data-width=500}

```{r}
gosBP <- params$goUp[params$goUp$Ont == "BP",]
plotGO(enrichdf = gosBP[params$bpnrup, ], nrows = length(params$bpnrup), ont="BP")

```

### GO dotplot --Upregulated genes-- {data-width=500}

```{r}
gosBP <- params$goUp[params$goUp$Ont == "BP",]
dotPlotGO(gosBP[params$bpnrup,], n = length(params$bpnrup))
```


Molecular function {data-navmenu="GO Upregulated"}
=====================================

Row {data-height=250}
------------------------------------

### Gene Onthology (GO) : Molecular function   {data-width=500}

Annotation related to the **molecular function (MF)**: Molecular-level activities performed by gene products. Describe activities that occur at the molecular level, such as “catalysis” or “transport”. GO molecular function terms represent activities rather than the entities (molecules or complexes) that perform the actions, and do not specify where, when, or in what context the action takes place. Molecular functions generally correspond to activities that can be performed by individual gene products (i.e. a protein or RNA), but some activities are performed by molecular complexes composed of multiple gene products.

Row {data-height=500}
------------------------------------

### GO Enrichment results --Upregulated genes-- {data-width=500}

```{r}
goDTup <- params$goDTup
names(goDTup)[names(goDTup) == "level"] <- "Ont.level"
    goDTup$Ont.level = as.integer(goDTup$Ont.level) 
    datatable2(goDTup[goDTup$Ont=="MF",], vars = c("genes"),
               filter = list(position="top", clear=FALSE),
               escape = FALSE,
               opts = list(pageLength = 10, white_space = "normal")
    )
```

### Texto variable {data-width=500}

```{r results='asis'}


```

Row {data-height=500}
------------------------------------

### GO barplot --Upregulated genes-- {data-width=500}

```{r}
gosMF <- params$goUp[params$goUp$Ont == "MF",]
plotGO(enrichdf = gosMF[params$mfnrup, ], nrows = length(params$mfnrup), ont="MF")

```

### GO dotplot --Upregulated genes-- {data-width=500}

```{r}
gosMF <- params$goUp[params$goUp$Ont == "MF",]
dotPlotGO(gosMF[params$mfnrup,], n = length(params$mfnrup))
```


Cellular component {data-navmenu="GO Upregulated"}
=====================================

Row {data-height=250}
------------------------------------

### Gene Onthology (GO) : Cellular component   {data-width=500}

Annotation related to the **celular component (CC):** The locations relative to cellular structures in which a gene product performs a function, either cellular compartments or stable macromolecular complexes of which they are parts.

Row {data-height=500}
------------------------------------

### GO Enrichment results --Upregulated genes-- {data-width=500}

```{r}
goDTup <- params$goDTup
names(goDTup)[names(goDTup) == "level"] <- "Ont.level"
    goDTup$Ont.level = as.integer(goDTup$Ont.level) 
    datatable2(goDTup[goDTup$Ont=="CC",], vars = c("genes"),
               filter = list(position="top", clear=FALSE),
               escape = FALSE,
               opts = list(pageLength = 10, white_space = "normal")
    )
```

### Texto variable {data-width=500}

```{r results='asis'}


```

Row {data-height=500}
------------------------------------

### GO barplot --Upregulated genes-- {data-width=500}

```{r}
gosCC <- params$goUp[params$goUp$Ont == "CC",]
plotGO(enrichdf = gosCC[params$ccnrup, ], nrows = length(params$ccnrup), ont="CC")

```

### GO dotplot --Upregulated genes-- {data-width=500}

```{r}
gosCC <- params$goUp[params$goUp$Ont == "CC",]
dotPlotGO(gosCC[params$ccnrup,], n = length(params$ccnrup))
```

<!-- DownRegulated BP-->

Biological Process {data-navmenu="GO Downregulated"}
=====================================

Row {data-height=250}
------------------------------------

### Gene Onthology (GO) : Biological process   {data-width=500}

Annotation related to the **biological process (BP)**: The larger processes, or ‘biological programs’ accomplished by multiple molecular activities. Note that a biological process is not equivalent to a pathway. At present, the GO does not try to represent the dynamics or dependencies that would be required to fully describe a pathway.


Row {data-height=500}
------------------------------------

### GO Enrichment results --Downregulated genes-- {data-width=500}

```{r}
goDTdown <- params$goDTdown
names(goDTdown)[names(goDTdown) == "level"] <- "Ont.level"
    goDTdown$Ont.level = as.integer(goDTdown$Ont.level) 
    datatable2(goDTdown[goDTdown$Ont=="BP",], vars = c("genes"),
               filter = list(position="top", clear=FALSE),
               escape = FALSE,
               opts = list(pageLength = 10, white_space = "normal")
    )
```

### Texto variable {data-width=500}

```{r results='asis'}


```

Row {data-height=500}
------------------------------------

### GO barplot --Downregulated genes-- {data-width=500}

```{r}
gosBP <- params$goDown[params$goDown$Ont == "BP",]
plotGO(enrichdf = gosBP[params$bpnrdown, ], nrows = length(params$bpnrdown), ont="BP")

```

### GO dotplot --Downregulated genes-- {data-width=500}

```{r}
gosBP <- params$goDown[params$goDown$Ont == "BP",]
dotPlotGO(gosBP[params$bpnrdown,], n = length(params$bpnrdown))
```


Molecular function {data-navmenu="GO Downregulated"}
=====================================

Row {data-height=250}
------------------------------------

### Gene Onthology (GO) : Molecular function   {data-width=500}

Annotation related to the **molecular function (MF)**: Molecular-level activities performed by gene products. Describe activities that occur at the molecular level, such as “catalysis” or “transport”. GO molecular function terms represent activities rather than the entities (molecules or complexes) that perform the actions, and do not specify where, when, or in what context the action takes place. Molecular functions generally correspond to activities that can be performed by individual gene products (i.e. a protein or RNA), but some activities are performed by molecular complexes composed of multiple gene products.

Row {data-height=500}
------------------------------------

### GO Enrichment results --Downregulated genes-- {data-width=500}

```{r}
goDTdown <- params$goDTdown
names(goDTdown)[names(goDTdown) == "level"] <- "Ont.level"
    goDTdown$Ont.level = as.integer(goDTdown$Ont.level) 
    datatable2(goDTdown[goDTdown$Ont=="MF",], vars = c("genes"),
               filter = list(position="top", clear=FALSE),
               escape = FALSE,
               opts = list(pageLength = 10, white_space = "normal")
    )
```

### Texto variable {data-width=500}

```{r results='asis'}


```

Row {data-height=500}
------------------------------------

### GO barplot --Downregulated genes-- {data-width=500}

```{r}
gosMF <- params$goDown[params$goDown$Ont == "MF",]
plotGO(enrichdf = gosMF[params$mfnrdown, ], nrows = length(params$mfnrdown), ont="MF")

```

### GO dotplot --Downregulated genes-- {data-width=500}

```{r}
gosMF <- params$goDown[params$goDown$Ont == "MF",]
dotPlotGO(gosMF[params$mfnrdown,], n = length(params$mfnrdown))
```


Cellular component {data-navmenu="GO Downregulated"}
=====================================

Row {data-height=250}
------------------------------------

### Gene Onthology (GO) : Cellular component   {data-width=500}

Annotation related to the **celular component (CC):** The locations relative to cellular structures in which a gene product performs a function, either cellular compartments or stable macromolecular complexes of which they are parts.

Row {data-height=500}
------------------------------------

### GO Enrichment results --Downregulated genes-- {data-width=500}

```{r}
goDTdown <- params$goDTdown
names(goDTdown)[names(goDTdown) == "level"] <- "Ont.level"
    goDTdown$Ont.level = as.integer(goDTdown$Ont.level) 
    datatable2(goDTdown[goDTdown$Ont=="CC",], vars = c("genes"),
               filter = list(position="top", clear=FALSE),
               escape = FALSE,
               opts = list(pageLength = 10, white_space = "normal")
    )
```

### Texto variable {data-width=500}

```{r results='asis'}


```

Row {data-height=500}
------------------------------------

### GO barplot --Downregulated genes-- {data-width=500}

```{r}
gosCC <- params$goDown[params$goDown$Ont == "CC",]
plotGO(enrichdf = gosCC[params$ccnrdown, ], nrows = length(params$ccnrdown), ont="CC")

```

### GO dotplot --Downregulated genes-- {data-width=500}

```{r}
gosCC <- params$goDown[params$goDown$Ont == "CC",]
dotPlotGO(gosCC[params$ccnrdown,], n = length(params$ccnrdown))
```


GSEA
=====================================


Row {data-height=250}
-------------------------------------

### Texto fijo {data-width=500}

Gene Set Enrichment Analysis (GSEA) is a computational method that determines whether an a priori defined set of genes shows statistically significant, concordant differences between two biological states (e.g. phenotypes). 


### Texto variable {data-width=500}

```{r}

```

Row {data-height=500}
------------------------------------

### GSEA  results {data-width=500}

```{r}
mygsea <- params$gsea
table <- mygsea@result[mygsea@result$p.adjust<=0.05 ,2:9] %>% 
      mutate_at(vars(3:7), ~round(., 4))
    DT::datatable( table,
                   rownames=FALSE,
                   filter = list(position="top", clear=FALSE),
                   options = list(
                     columnDefs = list(list(orderable = FALSE,
                                            className = "details-control",
                                            targets = 1)
                     ),
                     dom = "Bfrtipl",
                     buttons = c("copy", "csv", "excel", "pdf", "print"),
                     list(pageLength = 10, white_space = "normal")
                   )
    )
```


### Plot GSEA {data-width=500}

```{r}
enrichplot::gseaplot2(params$gsea, geneSetID = params$gseanr, pvalue_table = TRUE, ES_geom = "line")
```

About
===================================

Row
----------------------------------------------------

###

### Report template about {data-width=500}

#### **Enrich app 2020**

<br>

Thank you for using *Enrich app*

<br>

**Miriam Riquelme Pérez**

**Fernando Pérez Sanz**

```{r eval=TRUE, out.width="100px"}
knitr::include_graphics("tmpResources/dna-svg-small-13.gif")
```

For any suggestion or bug, please contact us

Corresponding author: miriam.riquelmep@gmail.com

###

