---
title: "RNAseq viewer report"
author: "`r params$author`"
output: 
    flexdashboard::flex_dashboard:
        vertical_layout: scroll
        orientation: rows
params:
  nr: NA
  bpnr: NA
  ccnr: NA
  mfnr: NA
  nrdown: NA
  bpnrdown: NA
  mfnrdown: NA
  ccnrdown: NA
  nrall: NA
  bpnrall: NA
  mfnrall: NA
  ccnrall: NA
  variablepca: NA
  tempdir: NA
  gseanr: NA
  author: "MRP"
---

```{r results="asis", echo=FALSE}
cat("
<style>
.navbar-inverse .navbar-nav>li>a {
    color: #ffffff;
    background-color: #334969;
}

.navbar-inverse .navbar-nav>.active>a, .navbar-inverse .navbar-nav>.active>a:hover, .navbar-inverse .navbar-nav>.active>a:focus {
    color: #ffffff;
    background-color: #3b5e86;
}

.navbar-inverse .navbar-collapse, .navbar-inverse .navbar-form {
    border-color: #396596;
    background-color: #334969;
}

.container-fluid {
    background-color: #334969;
}

.navbar-inverse .navbar-nav > .open > a, .navbar-inverse .navbar-nav > .open > a:hover, .navbar-inverse .navbar-nav > .open > a:focus {
    background-color: #3b5e86;
    color: #ffffff;

}

.dropdown-menu > .active > a, .dropdown-menu > .active > a:hover, .dropdown-menu > .active > a:focus {
    color: #ffffff;
    text-decoration: none;
    outline: 0;
    background-color: #3b5e86;
}

.navbar-inverse .navbar-nav > li > a:hover, .navbar-inverse .navbar-nav > li > a:focus {

    color: #ffffff;
    background-color: #3b5e86;
}

.navbar-inverse .navbar-nav > .open > a, .navbar-inverse .navbar-nav > .open > a:hover, .navbar-inverse .navbar-nav > .open > a:focus {

    background-color: #3b5e86;
    color: #ffffff;
}
.dropdown-menu > li > a:hover, .dropdown-menu > li > a:focus {
    text-decoration: none;
    color: #ffffff;
    background-color: #3b5e86;
}

#section-1.section.level3.chart-wrapper.chart-wrapper-flex{
    background: #f2f2f2;
    border: #f2f2f2
}
#section.section.level3.chart-wrapper.chart-wrapper-flex{
    background: #f2f2f2;
    border: #f2f2f2
}
#enrich-app-2020.section.level4{
    text-align: center;
}
</style>
")
```

```{r setup, echo=FALSE, include=FALSE, comment=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, fig.width = 12, eval=FALSE)

library('dygraphs')
library('flexdashboard')
library('shinydashboard')
library('limma')
library('tidyverse')
library('DT')
library('purrr')
library('plotly')
library('DESeq2')
library("ggraph")
library("igraph")
source("utils.R")
```

```{r}
#setwd(params$tempdir)
genesUp <- readRDS( "tmpResources/genesUp.Rds")
genesDown <- readRDS("tmpResources/genesDown.Rds")
genesall <- readRDS("tmpResources/kggDTall.Rds")
kggUp <- readRDS( "tmpResources/kggUp.Rds")
kggDTup <- readRDS( "tmpResources/kggDTup.Rds")
goUp <- readRDS( "tmpResources/goUp.Rds")
goDTup <- readRDS( "tmpResources/goDTup.Rds")
kggDown <- readRDS( "tmpResources/kggDown.Rds")
kggDTdown <- readRDS( "tmpResources/kggDTdown.Rds")
goDown <- readRDS( "tmpResources/goDown.Rds")
goDTdown <- readRDS( "tmpResources/goDTdown.Rds")
deseq <- readRDS("tmpResources/dds.Rds")
gsea <- readRDS("tmpResources/gsea.Rds")
kggAll <- readRDS( "tmpResources/kggAll.Rds")
kggDTall <- readRDS( "tmpResources/kggDTall.Rds")
```



Data preview
=====================================

Row {data-height=250}
--------------------------------------

### Biological context 

```{r results="asis"}
cat()
```

Row
--------------------------------------

### Metadata experiment {data-width=500}

```{r}

metadata <- as.data.frame(colData(deseq)) %>% select(-c(sizeFactor,replaceable))

datatable( metadata, extensions = "Buttons",
           rownames=FALSE,
           filter = list(position="top", clear=FALSE),
           options = list(
            columnDefs = list(list(orderable = TRUE,
                  className = "details-control",
                  targets = 1),
                  list(className = "dt-right", targets = 1:(ncol(metadata)-1))
                   ),
             dom = "Bfrtipl",
             buttons = c("copy", "csv", "excel", "pdf", "print"),
             list(pageLength = 10, white_space = "normal")
            )
)
```

### DEseq results {data-width=500}

```{r results="asis"}
res.sh <- as.data.frame(lfcShrink(deseq, coef=2, type="apeglm", res = results(deseq)))
conversion <- geneIdConverter(rownames(res.sh))
res.sh$baseMean <- round(res.sh$baseMean,4)
res.sh$lfcSE <- round(res.sh$lfcSE,4)
res.sh$log2FoldChange <- round(res.sh$log2FoldChange,4)
res.sh <- cbind(`Description`=conversion$description, res.sh)
res.sh <- cbind(`GeneName/Symbol`=conversion$consensus, res.sh)
res.sh <- res.sh[ ((res.sh$log2FoldChange >= logfc()[2] |
       res.sh$log2FoldChange< logfc()[1]) &
       res.sh$padj <= padj() ),]
res.sh <-  res.sh %>% select(-c(pvalue))
links = paste0("<a href='http://www.ensembl.org/Mus_musculus/Gene/Summary?db=core;g=",
       rownames(res.sh),"' target='_blank'>",rownames(res.sh),"</a>")
res.sh <- cbind(`GeneName/Ensembl`= links, res.sh)

datatable( res.sh, extensions = "Buttons", escape = FALSE, 
                  rownames = FALSE,
                  filter = list(position="top", clear=FALSE),
                  options = list(
                  columnDefs = list(list(orderable = FALSE,
                                         className = "details-control",
                                         targets = 1),
                                    list(className = "dt-right", targets = 1:ncol(res))
                                    ),
                  rowCallback = JS(
                                "function(row, data) {",
                                "for (i = 6; i < 8; i++) {",
                                "if (data[i]>1000 | data[i]<1){",
                                "$('td:eq('+i+')', row).html(data[i].toExponential(3));",
                                "}",
                                "}",
                                "}"),
                  dom = "Bfrtipl",
                  buttons = c("copy", "csv", "excel", "pdf", "print"),
                  list(pageLength = 10, white_space = "normal")
                  )
)
```

Row
--------------------------------------

### Samples PCA gene expression {data-width=100}

```{r}
plotPCA(datos$dds, intgroup = 'AAV' )+ #params$variablepca
            theme(plot.margin=unit(c(0.5,0.5,0.5,0.5),"cm"))+
            theme(text = element_text(size=20))
```


### Free text¿?¿ {data-width=50}

```{r out.width="800px", out.height="800px"}

```


All together {data-navmenu="Kegg"}
=====================================

Row {data-height=250}
------------------------------------

### Kyoto Encyclopedia of Genes and Genomes (KEGG) {data-width=500}

KEGG is a database resource for understanding high-level functions and utilities of the biological system, such as the cell, the organism and the ecosystem, from molecular-level information, especially large-scale molecular datasets generated by genome sequencing and other high-throughput experimental technologies.

Especially, KEGG PATHWAY is a collection of manually drawn pathway maps representing our knowledge on the molecular interaction, reaction and relation networks for:

* Metabolism:
    * Global/overview Carbohydrate Energy Lipid Nucleotide Amino acid Other amino Glycan
    * Cofactor/vitamin Terpenoid/PK Other secondary metabolite Xenobiotics Chemical structure
* Genetic Information Processing.
* Environmental Information Processing.
* Cellular Processes.
* Organismal Systems.
* Human Diseases.
* Drug Development.

### All genes {data-width=500}

```{r}
cat("#### Texto variable")
```


Row 
------------------------------------

### Kegg Enrichment results --All genes-- {data-width=500}

```{r echo=FALSE}
#predata <- kegg2DT(kggAll, genesall, nrows = NULL)
kgg <- kgg$all
    predata <- kggDT$all
    datatable2(
      predata,
      vars = c("genes"),
      filter = list(position="top", clear=FALSE),
      escape = FALSE,
      opts = list(pageLength = 10, white_space = "normal"))
```

### Barplot --all genes-- {data-width=500}

```{r  echo=FALSE}
plotKegg(enrichdf = kggUp[params$nrall,], nrows = length(params$nrall) ) 
```

Row
------------------------------------

### ChorPlot --all genes-- {data-width=500}

```{r echo=FALSE, warning=FALSE}
chordPlot(kggUp[params$nrall, ], nRows = length(params$nrall), orderby = "P.DE")
```


### Dotplot --all genes-- {data-width=500}

```{r}
dotPlotkegg(kggUp[params$nrall,], n = length(params$nrall))+
        theme(text = element_text(size=10))
```

Row
------------------------------------

### Kegg heatmap --all genes-- {data-width=500}

```{r}

```


### Kegg cnetplot --all genes-- {data-width=500}

```{r}

```


Upregulated genes {data-navmenu="Kegg"}
=====================================

Row {data-height=250}
------------------------------------

### Kyoto Encyclopedia of Genes and Genomes (KEGG) {data-width=500}

KEGG is a database resource for understanding high-level functions and utilities of the biological system, such as the cell, the organism and the ecosystem, from molecular-level information, especially large-scale molecular datasets generated by genome sequencing and other high-throughput experimental technologies.

Especially, KEGG PATHWAY is a collection of manually drawn pathway maps representing our knowledge on the molecular interaction, reaction and relation networks for:

* Metabolism:
    * Global/overview Carbohydrate Energy Lipid Nucleotide Amino acid Other amino Glycan
    * Cofactor/vitamin Terpenoid/PK Other secondary metabolite Xenobiotics Chemical structure
* Genetic Information Processing.
* Environmental Information Processing.
* Cellular Processes.
* Organismal Systems.
* Human Diseases.
* Drug Development.

### Upregulated genes {data-width=500}

```{r}
cat("#### Texto variable")
```


Row 
------------------------------------

### Kegg Enrichment results --Upregulated genes-- {data-width=500}

```{r echo=FALSE}

```

### Barplot --Upregulated genes-- {data-width=500}

```{r  echo=FALSE}

```

Row
------------------------------------

### ChorPlot --Upregulated genes-- {data-width=500}

```{r echo=FALSE, warning=FALSE}

```


### Dotplot --Upregulated genes-- {data-width=500}

```{r}

```

Row
------------------------------------

### Kegg heatmap --Upregulated genes-- {data-width=500}

```{r}

```


### Kegg cnetplot --Upregulated genes-- {data-width=500}

```{r}

```


Downregulated genes {data-navmenu="Kegg"}
=====================================

Row {data-height=250}
------------------------------------

### Kyoto Encyclopedia of Genes and Genomes (KEGG) {data-width=500}

KEGG is a database resource for understanding high-level functions and utilities of the biological system, such as the cell, the organism and the ecosystem, from molecular-level information, especially large-scale molecular datasets generated by genome sequencing and other high-throughput experimental technologies.

Especially, KEGG PATHWAY is a collection of manually drawn pathway maps representing our knowledge on the molecular interaction, reaction and relation networks for:

* Metabolism:
    * Global/overview Carbohydrate Energy Lipid Nucleotide Amino acid Other amino Glycan
    * Cofactor/vitamin Terpenoid/PK Other secondary metabolite Xenobiotics Chemical structure
* Genetic Information Processing.
* Environmental Information Processing.
* Cellular Processes.
* Organismal Systems.
* Human Diseases.
* Drug Development.

### Downregulated genes {data-width=500}

```{r}
cat("#### Texto variable")
```


Row 
------------------------------------

### Kegg Enrichment results --Downregulated genes-- {data-width=500}

```{r echo=FALSE}

```

### Barplot --Downregulated genes-- {data-width=500}

```{r  echo=FALSE}

```

Row
------------------------------------

### ChorPlot --Downregulated genes-- {data-width=500}

```{r echo=FALSE, warning=FALSE}

```


### Dotplot --Downregulated genes-- {data-width=500}

```{r}

```

Row
------------------------------------

### Kegg heatmap --Downregulated genes-- {data-width=500}

```{r}

```


### Kegg cnetplot --Downregulated genes-- {data-width=500}

```{r}

```



Biological Process {data-navmenu="GO all together"}
=====================================

Molecular function {data-navmenu="GO all together"}
=====================================

Cellular component {data-navmenu="GO all together"}
=====================================

Biological Process {data-navmenu="GO Upregulated"}
=====================================

Molecular function {data-navmenu="GO Upregulated"}
=====================================

Cellular component {data-navmenu="GO Upregulated"}
=====================================

Biological Process {data-navmenu="GO Downregulated"}
=====================================

Molecular function {data-navmenu="GO Downregulated"}
=====================================

Cellular component {data-navmenu="GO Downregulated"}
=====================================

GSEA
=====================================

Row
-------------------------------------

### Texto fijo {data-width=500}

```{r}

```


### Texto variable {data-width=500}

```{r}

```

Row
------------------------------------

### GSEA  results {data-width=500}

```{r}

```


### Plot GSEA {data-width=500}

```{r}

```

About
===================================

Row
----------------------------------------------------

###

### Report template about {data-width=500}

#### **Enrich app 2020**

<br>

Thank you for using *Enrich app*

<br>

**Miriam Riquelme Pérez**

**Fernando Pérez Sanz**

```{r eval=TRUE, out.width="100px"}
knitr::include_graphics("resources/dna-svg-small-13.gif")
```

For any suggestion or bug, please contact us

Corresponding author: miriam.riquelmep@gmail.com

###

