\PassOptionsToPackage{unicode=true}{hyperref} % options for packages loaded elsewhere
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provides euro and other symbols
\else % if luatex or xelatex
  \usepackage{unicode-math}
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage[]{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\usepackage{hyperref}
\hypersetup{
            pdftitle={RNAseq viewer report},
            pdfauthor={kk},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage[margin=1in]{geometry}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother

\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}

\title{RNAseq viewer report}
\author{kk}
\date{}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\hypertarget{biological-context}{%
\section{Biological context}\label{biological-context}}

\begin{itemize}
\tightlist
\item
  Samples background: CE1502 - Astrocytes WT with two different AAV (GFP
  vs SOCS3) extracted from the Kelly AD project.
\item
  Raw data (Fasta files):
  /local/aa\_SideProject\_ComparisonWT/FASTAfiles
\item
  Bam files

  \begin{itemize}
  \tightlist
  \item
    Aligned Genosplice (STAR 2.4): /home/Pactivastro/Genosplice-BAMfiles
    (All) or /local/aa\_SideProject\_ComparisonWT/BAMstar (subset)
  \end{itemize}
\end{itemize}

--outSAMstrandField intronMotif --outFilterMismatchNmax 2
--outFilterMultimapNmax 10 --outSAMunmapped Within --outSAMtype BAM
SortedByCoordinate --outStd BAM\_SortedByCoordinate --genomeLoad
NoSharedMemory --chimSegmentMin 15

\begin{itemize}
\tightlist
\item
  Aligned myself (STAR with customed genome adding GFPV5):
  /local/aa\_SideProject\_ComparisonWT/CUSTOM
\item
  Aligned myself (HISAT2):
  /local/aa\_SideProject\_ComparisonWT/BAMhisat2
\item
  Folder routes for the whole project

  \begin{itemize}
  \tightlist
  \item
    Local linux: /local/aa\_SideProject\_ComparisonWT (Full data files)
  \item
    Pactivastro:
    /home/Pactivastro/Resultats-manip/Miriam/aa\_SideProject\_ComparisonWT
    (Only reports)
  \end{itemize}
\end{itemize}

\hypertarget{metata-experiment}{%
\subsection{Metata experiment}\label{metata-experiment}}

\begin{table}[H]
\centering
\begin{tabular}{lrll}
\toprule
code & name & type & AAV\\
\midrule
\rowcolor[HTML]{ccffff}  Astro\_GFP\_AD\_63694 & 63694 & WT & GFP\\
Astro\_GFP\_WT\_63702 & 63702 & WT & GFP\\
\rowcolor[HTML]{ccffff}  Astro\_GFP\_WT\_63703 & 63703 & WT & GFP\\
Astro\_GFP\_WT\_63746 & 63746 & WT & GFP\\
\rowcolor[HTML]{ccffff}  Astro\_GFP\_WT\_63747 & 63747 & WT & GFP\\
\addlinespace
Astro\_GFP\_WT\_63765 & 63765 & WT & GFP\\
\rowcolor[HTML]{ccffff}  Astro\_GFP\_WT\_63778 & 63778 & WT & GFP\\
Astro\_SOCS3\_WT\_63704 & 63704 & WT & SOCS3\\
\rowcolor[HTML]{ccffff}  Astro\_SOCS3\_WT\_63750 & 63750 & WT & SOCS3\\
Astro\_SOCS3\_WT\_63780 & 63780 & WT & SOCS3\\
\addlinespace
\rowcolor[HTML]{ccffff}  Astro\_SOCS3\_WT\_63796 & 63796 & WT & SOCS3\\
Astro\_SOCS3\_WT\_63799 & 63799 & WT & SOCS3\\
\bottomrule
\end{tabular}
\end{table}

\hypertarget{deseq-result}{%
\subsection{DEseq result}\label{deseq-result}}

\begin{table}[H]
\centering
\begin{tabular}{lrrrrrr}
\toprule
Gene name & baseMean & log2FoldChange & lfcSE & stat & pvalue & padj\\
\midrule
\rowcolor{gray!6}  Col3a1 & 37.9402 & -7.9855 & 1.7999 & -4.4367 & 0e+00 & 0.0009\\
Slc40a1 & 1386.5615 & -1.5181 & 0.2624 & -5.7857 & 0e+00 & 0.0000\\
\rowcolor{gray!6}  Fn1 & 793.8706 & -2.9630 & 0.5910 & -5.0132 & 0e+00 & 0.0001\\
Sgpp2 & 314.5429 & -3.7740 & 0.6167 & -6.1200 & 0e+00 & 0.0000\\
\rowcolor{gray!6}  Sp100 & 257.6586 & -3.0534 & 0.5781 & -5.2820 & 0e+00 & 0.0000\\
\addlinespace
A630001G21Rik & 33.8002 & -8.3987 & 2.1373 & -3.9296 & 1e-04 & 0.0058\\
\rowcolor{gray!6}  Bok & 448.2461 & -2.0216 & 0.5429 & -3.7238 & 2e-04 & 0.0113\\
Mgat5 & 1876.3098 & 0.8007 & 0.2353 & 3.4025 & 7e-04 & 0.0309\\
\rowcolor{gray!6}  Btg2 & 3073.3391 & -0.9387 & 0.2134 & -4.3984 & 0e+00 & 0.0010\\
Ppfia4 & 243.4574 & -3.2151 & 0.9663 & -3.3272 & 9e-04 & 0.0386\\
\bottomrule
\end{tabular}
\end{table}

\hypertarget{pca-samples-plot}{%
\subsection{PCA samples plot}\label{pca-samples-plot}}

\end{document}
