---
title: "RNAseq viewer report"
author: "`r params$author`"
output: 
  pdf_document: 
    toc: yes
params:
  nr: NA
  bpnr: NA
  ccnr: NA
  mfnr: NA
  nrdown: NA
  bpnrdown: NA
  mfnrdown: NA
  ccnrdown: NA
  nrall: NA
  bpnrall: NA
  mfnrall: NA
  ccnrall: NA
  variablepca: NA
  tempdir: NA
  gseanr: NA
  author: NA
---

```{r echo=FALSE, include=FALSE, comment=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, fig.width = 12,
                      eval=TRUE, comment = FALSE, message = FALSE)
if("kableExtra" %in% .packages()){
  detach(package:kableExtra, unload = TRUE)
}
library('dygraphs')
library('flexdashboard')
library('shinydashboard')
library('limma')
library('tidyverse')
library('DT')
library('purrr')
library('plotly')
library('DESeq2')
library("ggraph")
library("igraph")
library("knitr")
library("kableExtra")
source("utils.R")
```

```{r}
setwd(params$tempdir)
# genesUp <- readRDS( "tmpResources/genesUp.Rds")
# genesDown <- readRDS("tmpResources/genesDown.Rds")
# genesall <- readRDS("tmpResources/kggDTall.Rds")
# kggUp <- readRDS( "tmpResources/kggUp.Rds")
# kggDTup <- readRDS( "tmpResources/kggDTup.Rds")
# goUp <- readRDS( "tmpResources/goUp.Rds")
# goDTup <- readRDS( "tmpResources/goDTup.Rds")
# kggDown <- readRDS( "tmpResources/kggDown.Rds")
# kggDTdown <- readRDS( "tmpResources/kggDTdown.Rds")
# goDown <- readRDS( "tmpResources/goDown.Rds")
# goDTdown <- readRDS( "tmpResources/goDTdown.Rds")
dds <- readRDS("tmpResources/dds.Rds")
# gsea <- readRDS("tmpResources/gsea.Rds")
# kggAll <- readRDS( "tmpResources/kggAll.Rds")
# kggDTall <- readRDS( "tmpResources/kggDTall.Rds")
```


# Biological context

```{r results="asis", eval=TRUE}

texto <- "
* Samples background: CE1502 - Astrocytes WT with two different AAV (GFP vs SOCS3) extracted from the Kelly AD project.
* Raw data (Fasta files): /local/aa_SideProject_ComparisonWT/FASTAfiles
* Bam files
    * Aligned Genosplice (STAR 2.4): /home/Pactivastro/Genosplice-BAMfiles (All) or /local/aa_SideProject_ComparisonWT/BAMstar (subset)

–outSAMstrandField intronMotif –outFilterMismatchNmax 2 –outFilterMultimapNmax 10 –outSAMunmapped Within –outSAMtype BAM SortedByCoordinate –outStd BAM_SortedByCoordinate –genomeLoad NoSharedMemory –chimSegmentMin 15

* Aligned myself (STAR with customed genome adding GFPV5): /local/aa_SideProject_ComparisonWT/CUSTOM
* Aligned myself (HISAT2): /local/aa_SideProject_ComparisonWT/BAMhisat2
* Folder routes for the whole project
    * Local linux: /local/aa_SideProject_ComparisonWT (Full data files)
    * Pactivastro: /home/Pactivastro/Resultats-manip/Miriam/aa_SideProject_ComparisonWT (Only reports)
"
cat(texto)
```

## Metata experiment

```{r eval=TRUE}

metadata <- as.data.frame(colData(dds)) %>% select(-c(sizeFactor,replaceable))
kable(metadata, "latex", booktabs = T, row.names = FALSE) %>%
    kable_styling(latex_options = "striped")
```

## DEseq result

```{r eval=FALSE}
res <- results(dds)
res <- as.data.frame(res)
res <- res[res$padj<=0.05, ]
conversion <- geneIdConverter(rownames(res))
res <- round(res,4)
res <- cbind(`Gene name`=conversion$consensus, res)
kable(head(res, 10), "latex", booktabs = T, row.names = FALSE) %>%
    kable_styling(latex_options = "striped" )
```

## plot PCA

```{r}
kk <- rlog(dds)
p <- plotPCA(kk, intgroup = "AAV")+
            theme(plot.margin=unit(c(0.5,0.5,0.5,0.5),"cm"))
```



