  ---
title: "RNAseq viewer report"
author: "`r params$author`"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  rmdformats::readthedown:
    highlight: kate
params:
  nr: NA
  bpnr: NA
  ccnr: NA
  mfnr: NA
  nrdown: NA
  bpnrdown: NA
  mfnrdown: NA
  ccnrdown: NA
  variablepca: NA
  tempdir: NA
  gseanr: NA
  author: NA
---


```{r results="asis", echo=FALSE}
cat("
<style>
#content {
  max-width: 1080px;
  margin-left: 300px;
  background: #edf0f2;
}
</style>
")
```

```{r setup, echo=FALSE, include=FALSE, comment=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, fig.width = 12)
setwd(params$tempdir)
library("shinydashboard")
library("limma")
library("tidyverse")
library("DT")
library("purrr")
library("plotly")
library("DESeq2")
library("ggraph")
library("igraph")
source("utils.R")
library("edgeR")
library("PoiClaClu")
library("ggplot2")
library("pheatmap")
library("RColorBrewer")
library("knitr")
library("apeglm")
library("calibrate")
library("AnnotationDbi")
library('org.Mm.eg.db')
library('AnnotationHub')
library('plotly')
library('topGO')
library('Rgraphviz')
library("BiocParallel")
library('clusterProfiler')
library('pathview')
library('gage')
library('ggpubr')
library('gageData')
library('EnsDb.Mmusculus.v79')
library('EnhancedVolcano')
library('gplots')

dds <- readRDS("tmpResources/deseq.Rds")

genesUp <- readRDS("tmpResources/genesUp.Rds")
genesDown <- readRDS("tmpResources/genesDown.Rds")

#genesAll <- readRDS()

kggUp <- readRDS("tmpResources/kggUp.Rds")
kggDTup <- readRDS("tmpResources/kggDTup.Rds")
kggDown <- readRDS("tmpResources/kggDown.Rds")
kggDTdown <- readRDS("tmpResources/kggDTdown.Rds")

#kggDTAll <- readRDS()

goUp <- readRDS("tmpResources/goUp.Rds")
goDTup <- readRDS("tmpResources/goDTup.Rds")
goDown <- readRDS("tmpResources/goDown.Rds")
goDTdown <- readRDS("tmpResources/goDTdown.Rds")

#goDTAll <- readRDS()

gsea <- readRDS("tmpResources/gsea.Rds")
```

# Data preview

## Metadata experiment

```{r}
metadata <- as.data.frame(colData(dds))
metadata$sizeFactor <- round(metadata$sizeFactor,4)
datatable(metadata, filter = list(position="top", clear=FALSE),
           options = list(
               columnDefs = list(list(orderable = FALSE,
               className = "details-control",
               targets = 1),
               list(className = "dt-right", targets = 1:ncol(metadata))),
               dom = "Bfrtipl",
               buttons = c("copy", "csv", "excel", "pdf", "print"),
               list(pageLength = 10, white_space = "normal")))
```

## PCA gene expresiÃ³n

```{r}

plotPCA(vst(dds), intgroup = params$variablepca, ntop = 500) +
            theme(plot.margin=unit(c(0.5,0.5,0.5,0.5),"cm")) +
            theme(text = element_text(size=12))

```

## Heatmaps

```{r echo=FALSE, fig.show='hold', out.width = "50%", fig.align = "default"}

sampleDists_vsd <- dist(t(assay(vst(dds))))

sampleDistMatrix_vsd <- as.matrix( sampleDists_vsd )
rownames(sampleDistMatrix_vsd) <- paste( dds$name, dds$AAV, sep = " - " )
colnames(sampleDistMatrix_vsd) <- paste( dds$name, dds$AAV, sep = " - " )
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix_vsd,
         clustering_distance_rows = sampleDists_vsd,
         clustering_distance_cols = sampleDists_vsd,
         col = colors, main = 'heatmap clustering in vsd data')


topVarGenes <- head(order(rowVars(assay(vst(dds))), decreasing = TRUE), 25)
mat  <- assay(vst(dds))[ topVarGenes, ]
mat  <- mat - rowMeans(mat)

df = data.frame(AAV = dds$AAV)
rownames(df) <- (colnames(mat))
colors <-  c("#008000","#800080")
names(colors) <- unique(df$AAV)

ann_color <- list(AAV = colors)

pheatmap(mat, cellwidth= 10.5, cellheight = 8.5, cluster_rows=TRUE, show_colnames=TRUE, cluster_cols=TRUE, annotation_col = df, labels_col = paste( dds$name, dds$AAV, sep = " - " ), annotation_colors = ann_color)

```


## Gene Expression results

```{r}

deseq <- DESeq(dds)

res <- results(deseq)
res.sh <- lfcShrink(deseq, coef="AAV_SOCS3_vs_GFP", type="apeglm", res = res)
res.sh <- as.data.frame(res.sh)
datatable(
    round(res.sh, 4),
    filter = list(position = "top", clear = FALSE),
    options = list(
        columnDefs = list(
            list(
                orderable = FALSE,
                className = "details-control",
                targets = 1
            ),
            list(className = "dt-right", targets = 1:ncol(res))
        ),
        dom = "Bfrtipl",
        buttons = c("copy", "csv", "excel", "pdf", "print"),
        list(pageLength = 10, white_space = "normal")
    )
)
```

## Volcano plot

## MA plot




# Kegg pathways enrichment

## Kegg enrichment Upregulated {.tabset .tabset-fade .tabset-pills}

```{r echo=FALSE}
predata <- kegg2DT(kggUp[params$nr, ], genesUp, nrows = length(params$nr))
datatable2( predata,
            vars = c("genes"),
            escape = FALSE,
            opts = list(pageLength = 10, white_space = "normal"))
```

### Kegg barplot

```{r  echo=FALSE}
plotKegg(enrichdf = kggUp[params$nr,], nrows = length(params$nr) ) 
```

### Kegg chordplot

```{r echo=FALSE, warning=FALSE}
chordPlot(kggUp[params$nr, ], nRows = length(params$nr), orderby = "P.DE")
```

### Kegg dotplot

```{r}
dotPlotkegg(kggUp[params$nr,], n = length(params$nr))+
        theme(text = element_text(size=10))
```

### Kegg heatmap

```{r}
heatmapKegg(kggDTup, params$nr)
```

### Kegg cnetplot

```{r}
customCnetKegg(kggUp, params$nr)
```


## Kegg enrichment Downregulated {.tabset .tabset-fade .tabset-pills}

```{r echo=FALSE}
predata <- kegg2DT(kggDown[params$nrdown, ], genesDown, nrows = length(params$nrdown))
datatable2( predata,
            vars = c("genes"),
            escape = FALSE,
            opts = list(pageLength = 10, white_space = "normal"))
```

### Kegg barplot

```{r  echo=FALSE}
plotKegg(enrichdf = kggDown[params$nrdown,], nrows = length(params$nrdown) ) 
```

### Kegg chordplot

```{r echo=FALSE, warning=FALSE}
chordPlot(kggDown[params$nrdown, ], nRows = length(params$nrdown), orderby = "P.DE")
```

### Kegg dotplot

```{r}
dotPlotkegg(kggDown[params$nrdown,], n = length(params$nrdown))+
        theme(text = element_text(size=10))
```

### Kegg heatmap

```{r}
heatmapKegg(kggDTdown, params$nrdown)
```

### Kegg cnetplot

```{r}
customCnetKegg(kggDown, params$nrdown)
```

# Gene Ontology enrichment

## GO Biological Proccess Upregulated {.tabset .tabset-fade .tabset-pills}

```{r echo=FALSE, eval=TRUE}
gosbp <- goUp[goUp$Ont=="BP",]
goDT <- go2DT(enrichdf = gosbp[params$bpnr, ], data = genesUp, nrows = length(params$bpnr))
datatable2(goDT, vars = c("genes"),
           escape = FALSE,
           opts = list(pageLength = 10, white_space = "normal"))
```

### BP barplot 

```{r echo=FALSE, eval=TRUE}
plotGO(enrichdf = gosbp[params$bpnr,], nrows = length(params$bpnr), ont="BP")
```

### BP dotplot

```{r}
dotPlotGO(gosbp[params$bpnr,], n = length(params$bpnr))+
        theme(text = element_text(size=10))
```

## GO Molecular Function Upregulated {.tabset .tabset-fade .tabset-pills}

```{r echo=FALSE, eval=TRUE}
gosbp <- goUp[goUp$Ont=="MF",]
goDT <- go2DT(enrichdf = gosbp[params$mfnr, ], data = genesUp, nrows = length(params$mfnr))
datatable2(goDT, vars = c("genes"),
           escape = FALSE,
           opts = list(pageLength = 10, white_space = "normal"))
```

### MF barplot

```{r echo=FALSE, eval=TRUE}
plotGO(enrichdf = gosbp[params$mfnr,], nrows = length(params$mfnr), ont="MF")
```

### MF dotplot

```{r}
dotPlotGO(gosbp[params$mfnr,], n = length(params$mfnr))+
        theme(text = element_text(size=10))
```

## GO Cellular Component Upregulated {.tabset .tabset-fade .tabset-pills}

```{r echo=FALSE, eval=TRUE}
gosbp <- goUp[goUp$Ont=="CC",]
goDT <- go2DT(enrichdf = gosbp[params$ccnr, ], data = genesUp, nrows = length(params$ccnr))
datatable2(goDT, vars = c("genes"),
           escape = FALSE,
           opts = list(pageLength = 10, white_space = "normal"))
```

### CC barplot

```{r echo=FALSE, eval=TRUE}
plotGO(enrichdf = gosbp[params$ccnr,], nrows = length(params$ccnr), ont="CC")
```

### CC dotplot

```{r}
dotPlotGO(gosbp[params$ccnr,], n = length(params$ccnr))+
        theme(text = element_text(size=10))
```


## GO Biological Proccess Downregulated {.tabset .tabset-fade .tabset-pills}

```{r echo=FALSE, eval=TRUE}
gosbp <- goDown[goDown$Ont=="BP",]
goDT <- go2DT(enrichdf = gosbp[params$bpnrdown, ],
              data = genesDown, nrows = length(params$bpnrdown))
datatable2(goDT, vars = c("genes"),
           escape = FALSE,
           opts = list(pageLength = 10, white_space = "normal"))
```

### BP barplot 

```{r echo=FALSE, eval=TRUE}
plotGO(enrichdf = gosbp[params$bpnrdown,], nrows = length(params$bpnrdown), ont="BP")
```

### BP dotplot

```{r}
dotPlotGO(gosbp[params$bpnrdown,], n = length(params$bpnrdown))+
        theme(text = element_text(size=10))
```

## GO Molecular Function Upregulated {.tabset .tabset-fade .tabset-pills}

```{r echo=FALSE, eval=TRUE}
gosbp <- goDown[goDown$Ont=="MF",]
goDT <- go2DT(enrichdf = gosbp[params$mfnrdown, ],
              data = genesDown, nrows = length(params$mfnrdown))
datatable2(goDT, vars = c("genes"),
           escape = FALSE,
           opts = list(pageLength = 10, white_space = "normal"))
```

### MF barplot

```{r echo=FALSE, eval=TRUE}
plotGO(enrichdf = gosbp[params$mfnrdown,], nrows = length(params$mfnrdown), ont="MF")
```

### MF dotplot

```{r}
dotPlotGO(gosbp[params$mfnrdown,], n = length(params$mfnrdown))+
        theme(text = element_text(size=10))
```

## GO Cellular Component Upregulated {.tabset .tabset-fade .tabset-pills}

```{r echo=FALSE, eval=TRUE}
gosbp <- goDown[goDown$Ont=="CC",]
goDT <- go2DT(enrichdf = gosbp[params$ccnrdown, ],
              data = genesDown, nrows = length(params$ccnrdown))
datatable2(goDT, vars = c("genes"),
           escape = FALSE,
           opts = list(pageLength = 10, white_space = "normal"))
```

### CC barplot

```{r echo=FALSE, eval=TRUE}
plotGO(enrichdf = gosbp[params$ccnrdown,], nrows = length(params$ccnrdown), ont="CC")
```

### CC dotplot

```{r}
dotPlotGO(gosbp[params$ccnrdown,], n = length(params$ccnrdown))+
        theme(text = element_text(size=10))
```

# Gene Set Enrichment Analysis

## GSEA results

```{r}
table <- gsea@result[gsea@result$p.adjust <= 0.05 , 2:9] %>%
  mutate_at(vars(3:7), ~ round(., 3))
DT::datatable(
  table,
  rownames = FALSE,
  filter = list(position = "top", clear = FALSE),
  options = list(
    columnDefs = list(
      list(
        orderable = FALSE,
        className = "details-control",
        targets = 1
      )
    ),
    dom = "Bfrtipl",
    buttons = c("copy", "csv", "excel", "pdf", "print"),
    list(pageLength = 10, white_space = "normal")
  )
)
```

## GSEA plot

```{r fig.width=12}
enrichplot::gseaplot2(gsea, geneSetID = params$gseanr, pvalue_table = TRUE, ES_geom = "line")
```

